inspire_disambiguation/core/ml/models.py

```diff
@@ -319,68 +585,118 @@ def __init__(self, estimator):
-                    if signature_uuid not in signatures_by_uuid:
-                        continue  # TODO figure out how this can happen
-                    self.X[i, 0] = signatures_by_uuid[signature_uuid]
-                    self.y[i] = cluster['cluster_id']
-                    i += 1
+        for cluster in input_clusters:
+            for signature_uuid in cluster["signature_uuids"]:
+                if signature_uuid not in signatures_by_uuid:
+                    continue  # TODO figure out how this can happen
+                self.X[i, 0] = signatures_by_uuid[signature_uuid]
+                self.y[i] = cluster["cluster_id"]
+                i += 1
 
     def load_model(self, input_filename):
-        with open(input_filename, 'rb') as fd:
+        """Loads model dumped by pickle
+
+        Args:
+            input_filename (str): path to file with dumped ethnicity model.
+        """
+        with open(input_filename, "rb") as fd:
             self.clusterer = pickle.load(fd)
 
     def save_model(self, output_filename):
-        with open_file_in_folder(output_filename, 'wb') as fd:
+        """Dump object to a file
+
+        Args:
+            output_filename (str): Path where model will be dumped
+        """
+        with open_file_in_folder(output_filename, "wb") as fd:
             pickle.dump(self.clusterer, fd, protocol=pickle.HIGHEST_PROTOCOL)
 
     def fit(self, n_jobs=8):
+        """Fit data using the estimator"""
         self.clusterer = BlockClustering(
             blocking=self.block_function,
             base_estimator=ScipyHierarchicalClustering(
                 affinity=_affinity,
                 threshold=self.clustering_threshold,
                 method=self.clustering_method,
-                supervised_scoring=b3_f_score),
+                supervised_scoring=b3_f_score,
+            ),
             n_jobs=n_jobs,
-            verbose=True)
+            verbose=True,
+        )
         self.clusterer.fit(self.X, self.y)
 
 
 @attr.s(slots=True)
 class Signature(object):
+    """Signature model for estimators"""
     author_affiliation = attr.ib()
     author_id = attr.ib()
     author_name = attr.ib()
     publication = attr.ib()
     signature_block = attr.ib()
     signature_uuid = attr.ib()
+    is_curated_author_id = attr.ib()
+
+    @classmethod
+    def build(cls, author, record):
+        """Build Signature object from author and record dictionaries
+
+        Args:
+            author (dict): dictionary containing author data
+            record (dict): dictionary containing record data
+
+        Returns:
+            Signature: Object built from provided data
+
+        """
+        author_id = _get_author_id(author)
+        return cls(
+            **{
+                "author_affiliation": _get_author_affiliation(author),
+                "author_id": author_id,
+                "author_name": author["full_name"],
+                "publication": Publication.build(record),
```

Comment:
If I remember correctly, previously the value was shared among all signatures in a given publication. Here you are creating a new publication for every signature, which might us significantly more memory.